solr/core/src/java/org/apache/solr/core/CoreContainer.java

```diff
@@ -2056,7 +2056,7 @@ public void reload(String name, UUID coreId) {
         newCore = core.reload(coreConfig);
 
         DocCollection docCollection = null;
-        if (getZkController() != null) {
+        if (getZkController() != null && !newCore.isSynthetic()) {
```

Comment:
maybe
```suggestion
        if (getZkController() != null && !(newCore instanceof SyntheticSolrCore)) {
```

Reply:
Thank you for the suggestion @cpoerschke ! 😊  I don't quite know the preference in general in the Solr dev community - I worked in some teams that avoid `instanceof` whenever possible.

I didn't ask for the full rationale behind that, my guess is probably due to the potential risk of missing future classes, which does not extend such target class yet inherit some similarity that fulfills the same condition. Having a `isXYZ` method could be slightly more flexible than forcing class hierarchy.

Anyway, I'm pretty okay with either way - whatever fits better for Solr community standards. The obvious benefit of using `instanceof` in this case is to avoid the extra `isSynthetic` method. 💪🏼